go/arrow/datatype_extension.go

```diff
@@ -122,18 +122,16 @@ type ExtensionType interface {
 	// If the storage type is incorrect or something else is invalid with the data this should
 	// return nil and an appropriate error.
 	Deserialize(storageType DataType, data string) (ExtensionType, error)
-
```

Comment:
please put the empty line back here, I'd prefer to keep a single empty line between the public and private method declarations here.